src/seml/config.py

```diff
@@ -254,7 +257,183 @@ def generate_configs(experiment_config, overwrite_params=None):
+
+        # Phase 1: Config updates
+        config_updates = convert_to_nested_dict(config)
+        distribute_config_updates(prefixes, scaffolding, config_updates)
+        
+        # Phase 2: Named Configs
+        for ncfg in named_config:
+            scaff, cfg_name = get_scaffolding_and_config_name(ncfg, scaffolding)
+            scaff.gather_fallbacks()
+            ncfg_updates = scaff.run_named_config(cfg_name)
+            distribute_presets(scaff.path, prefixes, scaffolding, ncfg_updates)
+            for ncfg_key, value in iterate_flattened(ncfg_updates):
+                set_by_dotted_path(config_updates, join_paths(scaff.path, ncfg_key), value)
+
+        distribute_config_updates(prefixes, scaffolding, config_updates)
+
+        # Phase 3: Normal config scopes
+        for scaffold in scaffolding.values():
+            scaffold.gather_fallbacks()
+            scaffold.set_up_config()
+
+            # update global config
+            config = get_configuration(scaffolding)
+            # run config hooks
+            config_hook_updates = scaffold.run_config_hooks(
+                config, exp.default_command, None
+            )
+            recursive_update(scaffold.config, config_hook_updates)
+
+        # Phase 4: finalize seeding
+        for scaffold in reversed(list(scaffolding.values())):
+            scaffold.set_up_seed()  # partially recursive
+
+        config_resolved = OmegaConf.to_container(OmegaConf.create(get_configuration(scaffolding)), resolve=True)
+        configs_resolved.append(remove_keys_from_nested(config_resolved, config_get_exclude_keys(config_resolved, config)))
+        
+    return configs_resolved
+   
+    
+def resolve_configs(executable: str, conda_env: str, configs: List[Dict], named_configs: List[List[str]], working_dir: str) -> List[Dict]:
+    """Resolves configurations by adding keys that are only added when the experiment is run to the MongoDB
+
+    Parameters
+    ----------
+    executable : str
+        Path to the executable
+    conda_env : str
+        Which conda environment to use
+    configs : List[Dict]
+        All experiment configurations
+    named_configs : List[str]
+        For each experiment, the named configurations to use.
+    working_dir : str
+        Which working directory to use
+
+    Returns
+    -------
+    List[Dict]
+        Resolved configurations
+    """
+    import sacred
+    exp_module = import_exe(executable, conda_env, working_dir)
+
+    # Extract experiment from module
+    exps = [v for k, v in exp_module.__dict__.items() if type(v) == sacred.Experiment]
+    if len(exps) == 0:
+        raise ExecutableError(f"Found no Sacred experiment. Something is wrong in '{executable}'.")
+    elif len(exps) > 1:
+        raise ExecutableError(f"Found more than 1 Sacred experiment in '{executable}'. "
+                              f"Can't check parameter configs. Disable via --no-sanity-check.")
```

Comment:
Error message adapted. Should resolving be optional? This could potentially lead to side-effects for the user if not taken care of properly. Named configs can also not be supported without resolving in the current state. What would be the use-case for not resolving?